helix-view/src/info.rs

```diff
@@ -41,6 +41,24 @@ impl Info {
         }
     }
 
+    pub fn from_string(title: &str, body: String) -> Self {
+        if body.is_empty() {
+            return Self {
+                title: title.to_string(),
+                height: 1,
+                width: title.len() as u16,
+                text: "".to_string(),
+            };
+        }
+
+        Self {
+            title: title.to_string(),
+            width: body.lines().map(|l| l.width()).max().unwrap() as u16,
+            height: body.lines().count() as u16,
+            text: body,
+        }
+    }
```

Comment:
Quite a bit of duplication with `fn new()`

Reply:
new method is needed because `Info::new` added some spaces so I added `Info::from_string` and refactored `Info::new`